torch/_decomp/decompositions_for_jvp.py

```diff
@@ -179,7 +179,7 @@ def native_layer_norm_backward(
         if len(outer_dim_indices) > 0:
             d_bias: Optional[Tensor] = torch.sum(grad_out, outer_dim_indices, False)
         else:
-            d_bias = grad_out
+            d_bias = grad_out.clone()
```

Comment:
This is the only thing that changed

Reply:
In combination with the changes in this PR, returning this input as-is as an output makes it so that we create a cycle in the graph.

Reply:
Can you explain the memory leak a bit more?

It would be concerning if we used decompositions for operators that didn't have forward-mode AD formulas by default and if this is an easy way to introduce memory leaks

Reply:
Suppose we have the function F(a) -> a (which just returns the input as-is) and we pass a leaf variable that that function, after the function completes we will have the cycle:

`a-> FBackward -> a's accumulate grad -> a`

This is why we don't allow the input tensor to be returned as-is for a non-composite functions (the TensorImpl use count checks should've helped with that here)

I don't think it is necessary for decompositions to follow ALL the same rules that a non-composite kernel should follow though. For example, the rules regarding views don't apply because the ADInplaceOverView is actually done after the decomposition so returning a view only sometimes should be OK.

Reply:
> Suppose we have the function F(a) -> a (which just returns the input as-is) and we pass a leaf variable that that function, after the function completes we will have the cycle:

Thanks for the explanation, that makes sense